bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -296,135 +332,166 @@ public void onValueReceived(@Nullable String variable, @Nullable String value, @
+                    }
 
-            if (thingTypeUID.equals(THING_TYPE_INSIGHT)) {
-                subscription = "insight1";
-                if (subscriptionState.get(subscription) != null) {
-                    logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
-                    service.removeSubscription(this, subscription);
+                    if (THING_TYPE_INSIGHT.equals(thingTypeUID)) {
+                        subscription = INSIGHTEVENT;
+                        if (subscriptionState.get(subscription) != null) {
+                            logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
+                            localService.removeSubscription(this, subscription);
+                        }
+                    }
+                    subscriptionState = new HashMap<>();
+                    localService.unregisterParticipant(this);
                 }
             }
-            subscriptionState = new HashMap<>();
-            service.unregisterParticipant(this);
-        }
-    }
-
-    private synchronized void onUpdate() {
-        ScheduledFuture<?> job = refreshJob;
-        if (job == null || job.isCancelled()) {
-            Configuration config = getThing().getConfiguration();
-            int refreshInterval = DEFAULT_REFRESH_INTERVALL_SECONDS;
-            Object refreshConfig = config.get("refresh");
-            if (refreshConfig != null) {
-                refreshInterval = ((BigDecimal) refreshConfig).intValue();
-            }
-            refreshJob = scheduler.scheduleWithFixedDelay(refreshRunnable, 0, refreshInterval, TimeUnit.SECONDS);
         }
     }
 
     private boolean isUpnpDeviceRegistered() {
-        return service.isRegistered(this);
+        UpnpIOService localService = service;
+        if (localService != null) {
+            return localService.isRegistered(this);
+        }
+        return false;
     }
 
     @Override
     public String getUDN() {
         return (String) this.getThing().getConfiguration().get(UDN);
     }
 
+    public String getHost() {
+        String localHost = host;
+        if (!localHost.isEmpty()) {
+            return localHost;
+        }
+        UpnpIOService localService = service;
+        if (localService != null) {
+            URL descriptorURL = localService.getDescriptorURL(this);
+            if (descriptorURL != null) {
+                return descriptorURL.getHost();
+            }
+        }
+        return localHost;
+    }
+
     /**
      * The {@link updateWemoState} polls the actual state of a WeMo device and
      * calls {@link onValueReceived} to update the statemap and channels..
      *
      */
     protected void updateWemoState() {
+        String actionService = BASICACTION;
+        String localhost = getHost();
+        if (localhost.isEmpty()) {
+            if (service != null) {
+                localhost = getHost();
```

Comment:
Same as previous comment, I don't think it makes sense to try twice.

Reply:
Changed in all handler classes